subiquity/common/filesystem/gaps.py

```diff
@@ -197,6 +197,8 @@ def maybe_add_gap(start, end, in_extended):
 def parts_and_gaps_disk(device, ignore_disk_fs=False):
     if device._fs is not None and not ignore_disk_fs:
         return []
+    if device.ptable == "unsupported":
+        return device.partitions()
```

Comment:
this or empty list?  I guess this is safe enough if the other manipulations actions are guarding against doing so on an unsupported ptable.

Reply:
Maybe it depends on context, parts_and_gaps has many call sites nowadays. But in general I would say `device.partitions()` is superior. It shows we acknowledge that there might be content on the disk potentially worth keeping. But if we run into problems with this approach, I'm inclined to try with an empty list.

Reply:
Works for me, please proceed.